### PR TITLE
Search plus

### PR DESCRIPTION
**Pull request by Hamza-777**
- Added user/post search functionality.
- Added comment and related functionality.
- Added edit comment/post functionality.

### CODE REVIEW

**Review by md-amir-gauhar — APPROVED** [review]
Looks good

File: src/Components/Starred/Starred.jsx [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
+import { getAllStarred } from '../../Reducers/userReducer';
 
 const Starred = () => {
+  const dispatch = useDispatch();
   const { starred } = useSelector((state) => state.user);
+  const { posts } = useSelector((state) => state.post);
+
+  useEffect(() => {
+    dispatch(getAllStarred());
+  }, [dispatch, posts]);
```
Comment (abhisheksahani-nova) [inline src/Components/Starred/Starred.jsx:13]: Suggestion :- Just adding posts in dependency array will work right ? What is the purpose of adding dispatch too

Comment (Hamza-777) [reply src/Components/Starred/Starred.jsx:13]: Just to get rid of console warnings.

Comment (Hamza-777) [reply src/Components/Starred/Starred.jsx:13]: Also can you please approve the changes?

**Review by abhisheksahani-nova — APPROVED** [review]
Looks clean and subtle , good to merge

File: src/Components/Starred/Starred.jsx (hunk 1 above)
Comment (abhisheksahani-nova) [reply src/Components/Starred/Starred.jsx:13]: Okay got it

**Pull request merged**